clients/fides-js/src/components/Tcf/TcfOverlay.tsx

```diff
@@ -0,0 +1,95 @@
+import { h, FunctionComponent } from "preact";
+import { useState, useCallback, useMemo } from "preact/hooks";
+import ConsentBanner from "../ConsentBanner";
+
+import { debugLog, hasActionNeededNotices } from "../../lib/consent-utils";
+
+import "../fides.css";
+import Overlay from "../Overlay";
+import { TcfConsentButtons } from "../ConsentButtons";
+import { OverlayProps } from "../types";
+import TcfTabs from "./TcfTabs";
+
+const TcfOverlay: FunctionComponent<OverlayProps> = ({
+  experience,
+  options,
+  cookie,
+}) => {
+  // TODO: how will we initialize TCF data?
```

Comment:
all of these TCF specific TODOs I figured should go in a separate PR since this refactor is already large